group_project_v2/utils.py

```diff
@@ -269,3 +269,14 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         self.field._values = self._old_values_value
         return False
+
+
+def add_resource(block, resource_type, path, fragment, via_url=False):
+    if via_url:
+        action = fragment.add_javascript_url if resource_type == 'javascript' else fragment.add_css_url
+        action_parameter = block.runtime.local_resource_url(block, path)
+    else:
+        action = fragment.add_javascript if resource_type == 'javascript' else fragment.add_css
+        action_parameter = loader.load_unicode(path)
+
+    action(action_parameter)
```

Comment:
@smarnach and it would defeat the whole point of having one function to manage adding resources. Those four one-liners should than be inlined, so instead of adding a function and switching all the code to using it I would have edit all the places where a resource is added. And those special cases (vendor js/css) would not be as clearly visible as they are now.